File: FRONTEND/lib/COMMON/Core/Models/Task Models/file_model.dart
import 'package:origami_structure/imports.dart';

class FileModel {
  String? image;
  String? username;
  String? firstName;
  String? lastName;
  String? taskTitle;
  String? taskFileName;
  String? taskBase64File;
  String? taskFile;
  double? taskFileSize;
  bool? focus;
  String? time;
  String? selectedEmoji;


  FileModel({
    this.image,
    this.username,
    this.firstName,
    this.lastName,
    this.taskTitle,
    this.taskFileName,
    this.taskBase64File,
    this.taskFile,
    this.taskFileSize,
    this.focus,
    this.time,
    this.selectedEmoji,
  });

  factory FileModel.fromJson(Map<String, dynamic> json) => FileModel(
    image: json["image"] ?? null,
    username:
    json["username"] ?? null,
    firstName:
    json["firstName"] ?? null,
    lastName:
    json["lastName"] ?? null,
    taskTitle:
    json["taskTitle"] ?? null,
    taskFileName: json["taskFileName"] ?? null,
    taskBase64File: json["taskBase64File"] ?? null,
    taskFile: json["taskFile"] ?? null,
    taskFileSize: json["taskFileSize"] == null ? null : json["taskFileSize"].toDouble(),
    focus: json["focus"] ?? null,
    time: json["time"] ?? null,
    selectedEmoji: json["selectedEmoji"] == null ? null : json["selectedEmoji"].toDouble(),
  );

  Map<String, dynamic> toJson() => {
    "image": image ?? null,
    "username": username ?? null,
    "firstName": firstName ?? null,
    "lastName": lastName ?? null,
    "taskTitle": taskTitle ?? null,
    "taskFileName": taskFileName ?? null,
    "taskBase64File": taskBase64File ?? null,
    "taskFile": taskFile ?? null,
    "taskFileSize": taskFileSize ?? null,
    "focus": focus ?? null,
    "time": time ?? null,
    "selectedEmoji": selectedEmoji ?? null,
  };
}

File: BACKEND/Models/Tasks Models/file_model.js
////////////////////////////////////////////// IMPORTS //////////////////////////////////////////////

// Import Mongoose
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

//////////////////////////////////////////// SCHEMA AND MODEL ////////////////////////////////////////////

// Create File Schema and Model
const FileSchema = new Schema({
    image: {
        type: String
    },
    username: {
        type: String
    },
    firstName: {
        type: String
    },
    lastName: {
        type: String
    },
    taskTitle: {
        type: String
    },
    taskFileName: {
        type: String
    },
    taskFile: {
        type: String
    },
    taskBase64File: {
        type: String
    },
    taskFileSize: {
        type: Number
    },
    focus: {
        type: Boolean
    },
    time: {
        type: String
    },
    selectedEmoji: {
        type: String
    },
});

//const CommentModel= mongoose.model('comment_data', CommentSchema, 'comment_data');
module.exports = FileSchema;